CMakeLists.txt

```diff
@@ -300,6 +295,11 @@ add_library(
         ${JNI_HEADER_DIR}/generated-headers.h
 )
 
+# We MUST set ACCP's RPATH to "$ORIGIN" so that the runtime dynamic loader will look in the same directory as ACCP for
+# any shared library dependencies BEFORE looking for a system-installed depencencies (such as libcrypto). If RPath is
+# not set, the loader may attempt to load an incompatible system-installed libcrypto and cause runtime crashes.
```

Comment:
I made that change locally, but reverted it since it didn't work as expected. The `libcrypto.so` file embeds a `SONAME` attribute within the ELF format. This `SONAME` attribute is what is actually used by the dynamic linker and loader, and the filename is ignored. Renaming `libcrypto.so` to `libamazonCorrettoCrypto.so` before the ACCP build began did not have an effect on the final libAACP shared library, which still declared a shared library dependency on `libcrypto.so`.

```
$ readelf -d libcrypto.so | grep SONAME
6: 0x000000000000000e (SONAME)             Library soname: [libcrypto.so]
```

Reply:
ah, interesting. that makes sense. thank you for the explanation.